x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_feature_control/draw_feature_control.tsx

```diff
@@ -71,7 +70,6 @@ export class DrawFeatureControl extends Component<Props, {}> {
         })
       );
     } finally {
```

Comment:
[DrawFeatureControl._onDraw] (https://github.com/elastic/kibana/blob/e047171fbed6c8dade96f48f8cd49e919a44a487/x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_feature_control/draw_feature_control.tsx#L45) has a bug because it does not await addNewFeatureToIndex. If there are multiple features (not possible) with the way the UI is set up then each feature will call addNewFeatureToIndex and the cursor will be set to SIMPLE_SELECT on the first feature completion.

To fix this problem, how about updating addNewFeatureToIndex to take an array of features. Then you can load each in the action and not have to worry about race conditions by calling addNewFeatureToIndex multiple times.

Reply:
I thought it wasn't a bug because it worked the same way in the main branch. But now I've changed the behavior, so that after adding a shape, the selected tool remains selected.

Reply:
> I thought it wasn't a bug because it worked the same way in the main branch. But now I've changed the behavior, so that after adding a shape, the selected tool remains selected.

I think there is some miscommunication. Keeping the draw shape selected is not what I was discussing.

Please revert commit 8c1620b7d4280d555e532f20f6f05117a1c1cd65

I am talking about the implemenation of DrawFeatureControl._onDraw. I have pasted below for clarity. Notice how the method uses `forEach` and calls `this.props.addNewFeatureToIndex` on each iteration. Instead, the method should use `forEach` to prepare an Geometry[] and call addNewFeatureToIndex once with a Geometry[].

```
_onDraw = async (e: { features: Feature[] }, mbDrawControl: MapboxDraw) => {
    try {
      e.features.forEach((feature: Feature) => {
        const { geometry } = geoJSONReader.read(feature);
        if (!geometry.isSimple() || !geometry.isValid()) {
          throw new Error(
            i18n.translate('xpack.maps.drawFeatureControl.invalidGeometry', {
              defaultMessage: `Invalid geometry detected`,
            })
          );
        }
        if ('coordinates' in feature.geometry) {
          // @ts-ignore /* Single position array only used if point geometry */
          const featureGeom: Geometry | Position[] =
            this.props.drawMode === DRAW_MODE.DRAW_POINTS
              ? feature.geometry.coordinates
              : feature.geometry;
          this.props.addNewFeatureToIndex(featureGeom);
        }
      });
    } catch (error) {
      getToasts().addWarning(
        i18n.translate('xpack.maps.drawFeatureControl.unableToCreateFeature', {
          defaultMessage: `Unable to create feature, error: '{errorMsg}'.`,
          values: {
            errorMsg: error.message,
          },
        })
      );
    } finally {
      this.props.disableDrawState();
      try {
        mbDrawControl.deleteAll();
      } catch (_e) {
        // Fail silently. Always works, but sometimes produces an upstream error in the mb draw lib
      }
    }
  };
```